packages/core/src/lib/nav-dropdown/nav-dropdown.svelte

```diff
@@ -0,0 +1,137 @@
+<script lang="ts">
+import { clickOutside } from '$lib/click-outside';
+import { Icon } from '$lib/icon';
+import { goto } from '$app/navigation';
+
+interface VersionOption {
+  label: string;
+  timeAgo?: string;
+  description?: string;
+  href: string;
+}
+
+export let options: VersionOption[] = [];
+
+let isOpen = false;
+
+const toggleDropdown = () => {
+  isOpen = !isOpen;
+};
+
+const closeDropdown = () => {
+  isOpen = false;
+};
+
+export let selectedOption: string | undefined = undefined;
+$: selected = selectedOption ?? options[0]?.label ?? 'Latest Version';
+
+let activeIndex = -1;
+
+const handleSelect = async (label: string, href: string) => {
+  selected = label;
+  closeDropdown();
+  await goto(href);
+};
+
+const handleKeydown = (event: KeyboardEvent) => {
+  switch (event.key) {
+    case 'Enter':
+    case ' ': {
+      if (!isOpen) {
+        event.preventDefault();
+        isOpen = true;
+      } else if (activeIndex >= 0) {
+        event.preventDefault();
+        const option = options[activeIndex];
+        if (option) {
+          void handleSelect(option.label, option.href);
+        }
+      }
+      break;
+    }
+    case 'Escape': {
+      if (isOpen) {
+        event.preventDefault();
+        closeDropdown();
+      }
+      break;
+    }
+    case 'ArrowDown': {
+      event.preventDefault();
+      if (isOpen) {
+        activeIndex = Math.min(activeIndex + 1, options.length - 1);
+      } else {
+        isOpen = true;
+      }
+      break;
+    }
+    case 'ArrowUp': {
+      event.preventDefault();
+      if (isOpen) {
+        activeIndex = Math.max(activeIndex - 1, 0);
+      } else {
+        isOpen = true;
+      }
+      break;
+    }
+  }
+};
+</script>
+
+<div
+  class="group/select-input relative flex w-48"
+  use:clickOutside={closeDropdown}
+>
+  <button
+    class="relative z-[2] h-7.5 w-full grow appearance-none border border-light bg-white py-1.5 pl-2 pr-1 text-xs leading-tight outline-none group-hover/select-input:border-gray-6"
+    on:click={toggleDropdown}
+    on:keydown={handleKeydown}
+    type="button"
+    aria-haspopup="menu"
+    aria-expanded={isOpen}
+  >
+    <div class="flex items-center justify-between">
+      <span class="block truncate text-xs">
+        {selected}
+      </span>
+      <Icon
+        name="chevron-down"
+        cx={['text-gray-6 transition-transform', { 'rotate-180': isOpen }]}
+      />
+    </div>
+  </button>
+
+  {#if isOpen}
+    <div
```

Comment:
The ticket mentioned to use <nav> element with <a> links, however, the linter was upset with me when I gave the nav element the role of menu.

Reply:
Ty!